tidy3d/components/geometry/mesh.py

```diff
@@ -347,25 +384,42 @@ def intersections_plane(
 
         mesh = self.trimesh
 
-        section = mesh.section(plane_origin=origin, plane_normal=normal)
-
-        if section is None:
-            return []
-
-        # homogeneous transformation matrix to map to xy plane
-        mapping = np.eye(4)
-
-        # translate to origin
-        mapping[3, :3] = -np.array(origin)
-
-        # permute so normal is aligned with z axis
-        # and (y, z), (x, z), resp. (x, y) are aligned with (x, y)
-        identity = np.eye(3)
-        permutation = self.unpop_axis(identity[2], identity[0:2], axis=axis)
-        mapping[:3, :3] = np.array(permutation).T
-
-        section2d, _ = section.to_planar(to_2D=mapping)
-        return list(section2d.polygons_full)
+        try:
+            section = mesh.section(plane_origin=origin, plane_normal=normal)
+
+            if section is None:
+                return []
+
+            # homogeneous transformation matrix to map to xy plane
+            mapping = np.eye(4)
+
+            # translate to origin
+            mapping[3, :3] = -np.array(origin)
+
+            # permute so normal is aligned with z axis
+            # and (y, z), (x, z), resp. (x, y) are aligned with (x, y)
+            identity = np.eye(3)
+            permutation = self.unpop_axis(identity[2], identity[0:2], axis=axis)
+            mapping[:3, :3] = np.array(permutation).T
+
+            section2d, _ = section.to_planar(to_2D=mapping)
+            return list(section2d.polygons_full)
+
+        except ValueError as e:
+            if not mesh.is_watertight:
+                log.warning(
+                    "Unable to compute 'TriangleMesh.intersections_plane' "
+                    "because the mesh was not watertight. Using bounding box instead. "
+                    "This may be overly strict; consider using 'TriangleMesh.fill_holes' "
+                    "to repair the non-watertight mesh."
+                )
+            else:
+                log.warning(
+                    "Unable to compute 'TriangleMesh.intersections_plane'. "
+                    "Using bounding box instead."
+                )
+            log.warning(e)
+            return self.bounding_box.intersections_plane(x=x, y=y, z=z)
```

Comment:
Did you manage to test this part of the code with a simulation that hits this error? And shouldn't there be a unit test for it? Also maybe instead of `log.warning(e)`, `log.warning(f"Error encountered: {e}")` (I think this works but maybe `str(e)` is needed?)

Reply:
Unfortunately, no -- even after playing around a bunch, I wasn't able to encounter this error. I think the trimesh library is also designed not to give this error, even for a wide range of meshes (e.g. even when not watertight, and even when this intersects the plane). I did improve the other tests a bit though.

On the bright side, although this isn't tested, it won't make the handling worse.